firmware-management/swupdate-handlers/rootfs-handler.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <errno.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include "swupdate.h"
+#include "util.h"
+#include "handler.h"
+#include "arm-handler-common.h"
+
+int copy_image_and_sync(struct img_type *img, const char *const device_filepath)
+{
+    int ret_val = 0;
+    int fd = openfileoutput(device_filepath);
+    if (fd < 0)
+    {
+        ERROR("%s", "Failed to open target device file");
+        return -1;
+    }
+
+    if (copyimage(&fd, img, NULL) < 0)
+    {
+        ERROR("%s", "Failed to copy image to target device");
+        goto close;
+        ret_val = -1;
+    }
+
+    if (fsync(fd) == -1)
+    {
+        WARN("%s: %s", "Failed to sync filesystem", strerror(errno));
+    }
+
+close:
+    if (close(fd) == -1)
+    {
+        ERROR("%s", "Failed to close target device file descriptor");
+        ret_val = -1;
+    }
+
+    return ret_val;
+}
+
+int get_bootflag_file_path(char *const bootflags_file_path, const size_t size)
+{
+    static const char *const fname = "/rootfs2";
+    const size_t len = strlen(BOOTFLAGS_DIR) + strlen(fname) + 1;
+    if (len > size)
+    {
+        ERROR("%s", "Bootflag file path will not fit in the destination buffer");
+        return -1;
+    }
+
+    return snprintf(bootflags_file_path, len, "%s%s", BOOTFLAGS_DIR, fname);
+}
+
+int write_bootflag_file()
```

Comment:
```suggestion
static int write_bootflag_file(void)
```

Reply:
Why do we need the `void`?

Reply:
Because in C (unlike C++), 
```C
static int write_bootflag_file()
```
doesn't declare that `write_bootflag_file` is a function that takes no parameters, it declares that `write_bootflag_file` is a function that takes some unknown number of parameters of unknown types...At least historically... Or at least that's what it means when it's just a declaration without a definition. With the definition I'm not as sure but 
```C
static int write_bootflag_file(void)
```
is how C programmers generally write it.

Reply:
Thanks for the explanation. I've changed this function to take a `filename` parameter and moved it to arm-handler-common (along with the other bootflag file related functions) as we may be able to reuse them.